python/helpers/lib/parser.py

```diff
@@ -226,25 +226,27 @@ def fake_open(*args, **kwargs):
 
     if os.path.isfile(setup_cfg_path):
         try:
-            config = setuptools.config.read_configuration(setup_cfg_path)
+            config = configparser.ConfigParser()
+            config.read(setup_cfg_path)
 
             for req_type in [
                 "setup_requires",
                 "install_requires",
                 "tests_require",
             ]:
-                requires = config.get("options", {}).get(req_type, [])
+                requires = config.get('options', req_type, fallback='').splitlines()
+                requires = [req for req in requires if req.strip()]
                 parse_requirements(requires, req_type, setup_cfg)
 
-            extras_require = config.get("options", {}).get(
-                "extras_require", {}
-            )
-            for key, value in extras_require.items():
-                parse_requirements(
-                    value, "extras_require:{}".format(key), setup_cfg
-                )
+            if config.has_section('options.extras_require'):
```

Comment:
@amazimbe : I took the liberty to make changes in your branch, hope thats ok. Executing native helper script provides below warning where `setuptools.config.read_configuration` is going to be deprecated so we have to anyways replace it in the future. The setuptools currently getting used is [v72.1.0](https://github.com/pypa/setuptools/releases/tag/v72.1.0) which was released 2 weeks ago and this is the issue. I have changed the native helper to use `config  parser` to to parse the setup.cfg keeping all the functionality same.

```bash
how to update the setuptools.config.read_configuration to use build.util.project_wheel_metadata for reading setup.cfg provided here. update the parser.py

/opt/python/lib/parser.py:229: SetuptoolsDeprecationWarning: Deprecated API usage.
!!

        ********************************************************************************
        As setuptools moves its configuration towards `pyproject.toml`,
        `setuptools.config.read_configuration` became deprecated.

        For the time being, you can use the `setuptools.config.setupcfg` module
        to access a backward compatible API, but this module is provisional
        and might be removed in the future.

        To read project metadata, consider using
        ``build.util.project_wheel_metadata`` (https://pypi.org/project/build/).
        For simple scenarios, you can also try parsing the file directly
        with the help of ``configparser``.
        ********************************************************************************
```

Reply:
@jurre ^^

Reply:
@honeyankit Yes, that is fine.